pkg/utils/kube/kube.go

```diff
@@ -135,6 +135,24 @@ func GetAppInstanceLabel(un *unstructured.Unstructured, key string) string {
 	return ""
 }
 
+// GetAppInstanceAnnotation returns the application instance name from annotations
+func GetAppInstanceAnnotation(un *unstructured.Unstructured, key string) string {
```

Comment:
This method can be just `un.GetAnnotations()[key]` :) It's safe to index into a `nil` map. https://play.golang.org/p/zvWU5a86Ry_f